alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/LicenseMetrics.java

```diff
@@ -0,0 +1,118 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.alfresco.service.descriptor.Descriptor;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.service.license.LicenseDescriptor;
+import org.alfresco.service.license.LicenseService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import javax.annotation.Nonnull;
+
+public class LicenseMetrics implements MeterBinder, ApplicationContextAware {
+
+    private static final Logger logger = LoggerFactory.getLogger(LicenseMetrics.class);
+
+    private static final String METRIC_NAME_LICENSE = "license";
+
+    private ApplicationContext ctx;
+    private DescriptorService descriptorService;
+
+    public LicenseMetrics(DescriptorService descriptorService) {
+        this.descriptorService = descriptorService;
+    }
+
+    @Override
+    public void bindTo(@Nonnull MeterRegistry registry) {
+        // do not do anything for Community
+        Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+        if(!"Enterprise".equals(serverDescriptor.getEdition())) {
+            logger.info("Edition={}, license metrics are not available", serverDescriptor.getEdition());
```

Comment:
Hmm, I'm thinking if we should -- for ease of monitoring -- always export these license metrics, also for community editions.

But instead of reporting actual values for community editions, we always report `valid=1,remainingDays=Integer.MAX_VALUE,maxDocs=Long.MAX_VALUE,maxUsers=Long.MAX_VALUE`

Reply:
What is the point of exporting things which don't exist? How does it help monitoring?
It might confuse some people (expecting a license for community) and it occupies space.

Reply:
Uniformity.

But I would use `-1` for `maxDocs` and `maxUsers` ?

Reply:
Using `-1` to mean `infinity` has its own set of problems when doing math on timeseries.
When creating an alert for document usage, you could do something like `predict_linear(current_docs[7d], '1 month') > max_docs` to alert approximately one month before max docs is reached. When it is -1, this alert will always fire.

Reply:
Uniformity for grafana? Is it better to have fake data than an empty panel? 
Alerts will not be trigerred anyway.

Reply:
Good point, I was thinking more about alerting than displaying data.

Maybe it's better to show empty panels in grafana indeed, showing "fake" data there may even be more confusing.